batch/batch/worker/worker.py

```diff
@@ -2231,6 +2240,11 @@ async def cleanup(self):
                 self.format_version, self.batch_id, self.job_id, self.attempt_id, 'main', log_contents
             )
 
+        if self.cloudfuse:
+            n_mounted = len([config for config in self.cloudfuse if config['mounted']])
+            if n_mounted > 0:
+                return
```

Comment:
I think it's worth outlining here what the danger is and adding a comment for defensive pieces of code like this. Otherwise code like this will pile up and we will forget what it was instituted for. I'm very likely to delete what I think is redundant or unnecessary code unless a comment warns me against that line of thinking.

Reply:
Additionally, we should institute a requirement that JVMJob cloudfuse mount destinations start with `/cloudfuse`, as that is the only mount point that we set as `slave` and could therefore be used in JVMJobs.

Reply:
Actually, apologies, I think bailing out early if an unmount fails is self-explanatory enough. I think what initially gave me pause is this happening in a JVMJob. What's the state of a JVM container where an unmount has failed? That means that another job that comes in could potentially see that mount? That seems really not great, I feel like we should kill the JVM container at that point and create a new one.